### PR TITLE
Add Safer Cluster with IAP bastion host access pattern

### PR DESCRIPTION
**Pull request by bharathkkb**
Demonstrates a realistic private IP access pattern for a private safer cluster.

Replaces #322

### CODE REVIEW

**Comment by bharathkkb** [comment]
@morgante PTAL

File: examples/safer_cluster_iap_bastion/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+3. Run `terraform apply`.
+
+4. After apply is complete, generate kubeconfig for the private cluster. _The command with the right parameters will displayed as the Terraform output `get_credentials_command`._
+
+   ```sh
+   gcloud container clusters get-credentials --project $PROJECT_ID --zone $ZONE --internal-ip $CLUSTER_NAME
+   ```
+
+5. SSH to the Bastion Host while port forwarding to the bastion host through an IAP tunnel. _The command with the right parameters will displayed as the Terraform output `bastion_ssh_command`._
```
Comment (morgante) [inline examples/safer_cluster_iap_bastion/README.md:23]:
```suggestion
5. SSH to the Bastion Host while port forwarding to the bastion host through an IAP tunnel. _The command with the right parameters will displayed by running `terraform output bastion_ssh_command`._
```

File: examples/safer_cluster_iap_bastion/variables.tf [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  default     = "safer-cluster-network"
+}
+
+variable "subnet_name" {
+  type        = string
+  description = "The name of the subnet being created to host the cluster in"
+  default     = "safer-cluster-subnet"
+}
+
+variable "subnet_region" {
```
Comment (morgante) [inline examples/safer_cluster_iap_bastion/variables.tf:47]: This is redundant, just use `var.region`

**Comment by bharathkkb** [comment]
@morgante I have added a simple test that ssh into the bastion and verifies connectivity to private cluster. Also looks like `sandbox-enabled` tests have been failing for unrelated reasons. Possible fix here: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/544/commits/7ea835beb62f380fa507eee410995e3af35f7091

File: examples/safer_cluster_iap_bastion/variables.tf (hunk 2 above)
Comment (bharathkkb) [reply examples/safer_cluster_iap_bastion/variables.tf]: done

**Pull request merged**